aws/data_source_aws_vpc.go

```diff
@@ -81,7 +81,15 @@ func dataSourceAwsVpcRead(d *schema.ResourceData, meta interface{}) error {
 
 	req := &ec2.DescribeVpcsInput{}
 
-	if id := d.Id(); id != "" {
+	// set id from state, if we have it
+	id := d.Id()
```

Comment:
This line is actually unnecessary, the ID will never be available from state or at least not until `d.SetId()` is called.

The state of a data source is always built from scratch on every refresh.

Reply:
My understanding was this line was valid for subsequent runs, is that not correct?

Reply:
There's no such thing as "subsequent" run for a data source - they're all the same as there's no existing state - try adding logging and you'll see it's always empty.

Reply:
👍